### PR TITLE
hotfix for frontend when z=Null

### PR DESCRIPTION
**Pull request by profjsb**
Interestingly, it looks like our test suite and demo data ALWAYS set a redshift (zero or otherwise) so we never tested z=Null until ... production. This fixes a few places where z=Null was throwing errors on the frontend.

### CODE REVIEW

File: static/js/components/Candidate.jsx [hunk 1]
```diff
@@ -92,8 +92,12 @@ const Candidate = ({ route }) => {
                 {candidate.gal_lat.toFixed(6)}
                 )
                 <br />
-                <b>Redshift: &nbsp;</b>
-                {candidate.redshift.toFixed(4)}
+                {candidate.redshift != null && (
```
Comment (dannygoldstein) [inline static/js/components/Candidate.jsx:95]:
```suggestion
                {candidate.redshift && (
```

File: static/js/components/SourceDesktop.jsx [hunk 2]
```diff
@@ -142,8 +142,12 @@ const SourceDesktop = ({ source }) => {
           {source.gal_lat.toFixed(6)}
           )
           <br />
-          <b>Redshift: &nbsp;</b>
-          {source.redshift.toFixed(4)}
+          {source.redshift != null && (
```
Comment (dannygoldstein) [inline static/js/components/SourceDesktop.jsx:145]:
```suggestion
          {source.redshift && (
```

File: static/js/components/SourceDesktop.jsx [hunk 3]
```diff
@@ -163,7 +167,7 @@ const SourceDesktop = ({ source }) => {
               &nbsp; Mpc
             </>
           )}
-          &nbsp;|&nbsp;
+          {source.redshift != null && <>&nbsp;|&nbsp;</>}
```
Comment (dannygoldstein) [inline static/js/components/SourceDesktop.jsx:170]:
```suggestion
          {source.redshift && <>&nbsp;|&nbsp;</>}
```

File: static/js/components/Candidate.jsx (hunk 1 above)
Comment (profjsb) [inline static/js/components/Candidate.jsx:95]:
```suggestion
                {(candidate.redshift != null) && (
```

File: static/js/components/SourceMobile.jsx [hunk 4]
```diff
@@ -165,8 +165,12 @@ const SourceMobile = ({ source }) => {
               {source.gal_lat.toFixed(1)}
               )
               <br />
-              <b>Redshift: &nbsp;</b>
-              {source.redshift.toFixed(4)}
+              {source.redshift != null && (
```
Comment (dannygoldstein) [inline static/js/components/SourceMobile.jsx:168]:
```suggestion
              {source.redshift && (
```

File: static/js/components/SourceMobile.jsx [hunk 5]
```diff
@@ -186,7 +190,7 @@ const SourceMobile = ({ source }) => {
                   &nbsp; Mpc
                 </>
               )}
-              &nbsp;|&nbsp;
+              {source.redshift != null && <>&nbsp;|&nbsp;</>}
```
Comment (dannygoldstein) [inline static/js/components/SourceMobile.jsx:193]:
```suggestion
              {source.redshift && <>&nbsp;|&nbsp;</>}
```

**Review by dannygoldstein — APPROVED** [review]

File: static/js/components/Candidate.jsx (hunk 1 above)
Comment (profjsb) [reply static/js/components/Candidate.jsx:95]: No that doesn't work. If z=0 then this doesn't pass

File: static/js/components/SourceDesktop.jsx (hunk 2 above)
Comment (profjsb) [reply static/js/components/SourceDesktop.jsx:145]: Nope...see above.

File: static/js/components/SourceDesktop.jsx (hunk 3 above)
Comment (profjsb) [reply static/js/components/SourceDesktop.jsx:170]: Nope

File: static/js/components/SourceMobile.jsx (hunk 4 above)
Comment (profjsb) [reply static/js/components/SourceMobile.jsx:168]: Nope

File: static/js/components/SourceMobile.jsx (hunk 5 above)
Comment (profjsb) [reply static/js/components/SourceMobile.jsx:193]: Nope

**Review by dannygoldstein — APPROVED** [review]
One more comment: can you add a test to the test suite? Object has null redshift, check that things render correctly on the frontend.

**Comment by dmitryduev** [comment]
Oops, my bad @dannygoldstein, didn't see your last comment. Could that be another quick PR?
